### ISSUE TITLE
[MenuButton] Testing 

### ISSUE THREAD

**Issue by davidicus** [carbon-design-system/carbon-addons-iot-react#2265]
MenuButton component needs comprehensive testing in a few areas. This issue is considered complete when:

- [x] data test ids have been provided to appropriate pieces 
- [x] Unit test are 80%+
- [ ] a11y scans in storybook do not show any violations 
- [x] a11y unit test has been written using the IBM scan tool
- [x] RTL has been tested and working properly. 
- [x] Confirm that all strings are able to be substituted for i18n

**Comment by kevinsperrine** [comment]
@davidicus How do we handle a11y testing/violations in this case where the violations are on the Carbon side? Do we open issues and submit PRs?
